### PR TITLE
fix(client): prevent iframe from duplicating by using preventDefault

### LINKED ISSUE #46057: Preview editor re-renders incorrectly
### Describe the Issue

On Firefox, clicking a link inside the preview pane and pressing 'back' on your web browser causes the preview pane to reload with the entire content of the page (html, css, preview pane etc.)
image

On Chrome, this issue is not present, BUT it still re-renders the preview pane incorrectly. Instead of being able to create multiple pages in one preview pane, Chrome renders blank content until the preview pane is closed and re-opened.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-57

### Steps to Reproduce

1. Go to the preview pane.
2. Click on a link (or create one in html first)
3. Press back in your browser.
4. Preview pane now contains the content of the page.
5. Repeat and create more panes.

### Expected behavior

I expected the page to maybe go blank or just go back to the last web page. I did not expect to be able to multiply the content of the page.

### Screenshots

image

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46057

This is another attempt with a different solution. 

Let me know if you guys like this flow.

### CODE REVIEW

**Comment by raisedadead** [comment]
Smart! This will possibly conflict with https://github.com/freeCodeCamp/freeCodeCamp/pull/46075, I think it's best we merge that first to help preserve your history later.

**Review by ojeytonwilliams — DISMISSED** [review]

File: client/src/templates/Challenges/utils/frame.ts [hunk 1]
```diff
@@ -69,7 +69,7 @@ const createHeader = (id = mainPreviewId) => `
       }
       if(element && element.nodeName === 'A'){
         e.preventDefault();
-        window.parent.window.alert('Normally this link would bring you to another website! It works!' \n + "(" + element.href + ")");
+        window.parent.window.alert('Normally this link would bring you to another website! It works!' + 'This is a link to: ' + '(' + element.href + ')');
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/utils/frame.ts:72]:
```suggestion
        window.parent.window.alert('Normally this link would bring you to another website! It works!' + ' This is a link to: ' + '(' + element.href + ')');
```

Spacing:

image

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Review by naomi-lgbt — APPROVED** [review]
Okay I'm done nitpicking :)

**Review by Nirajn2311 — COMMENTED** [review]
Neat solution by not allowing the user to change the page itself. 

But if the user submits the form and click back button then the same issue is there.

**Comment by Sembauke** [comment]
Good catch Niraj, it seems though that we can't use the same solution as for clicking links.

One of my speculations (might have read is somewhere) that an Iframe created with JavaScript does not work well with Firefox.

I propose to merge **this** without a solution for _that particular problem_ but keep the issue open and update it.

**Review by Nirajn2311 — APPROVED** [review]

**Comment by Nirajn2311** [comment]
Updated the issues and added scope before merging this PR to keep Mrugesh happy 😄

File: client/src/templates/Challenges/utils/frame.ts [hunk 2]
```diff
@@ -67,6 +67,10 @@ const createHeader = (id = mainPreviewId) => `
       while(element && element.nodeName !== 'A') {
         element = element.parentElement;
       }
+      if(element && element.nodeName === 'A'){
+        e.preventDefault();
+        window.parent.window.alert('Normally this link would bring you to another website! It works!' + ' This is a link to: ' + '(' + element.href + ')');
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/utils/frame.ts:72]: Are we wanting to translate this?

Comment (ojeytonwilliams) [reply client/src/templates/Challenges/utils/frame.ts:72]: Sounds good.  The build error message could use translating, too.

**Pull request merged**